Swift/tree.playground/Contents.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+
+
+class Node<T>: CustomStringConvertible where T: CustomStringConvertible, T: Comparable {
```

Comment:
That is not the preferred code style and is wrong in the means of your task. It is mainly used when you have several Generic and try to constrain them like:
```Swift
class Nodes<A, B, C> where A: Comparable, B: Equatable, C: CustomStringConvertible {
	
}
```
It may be better to use ``&``here.
```Swift
class Node<T>: CustomStringConvertible where T: CustomStringConvertible & Comparable {
```
because you use only one Generic and only a few protocols. When you have more protocols It is better to use a ``typealias``
```Swift
typealias NodeRelatedProtocol = CustomStringConvertible & Comparable & Equatable

class Node<T>: CustomStringConvertible where T: NodeRelatedProtocol {
```
Yet for this task you should use extensions combined with the element-based behavior.
```Swift
public class Node<T> {

	public typealias Element = T
	public let value: Element
	
	public init(with value: Element) {
		self.value = value
	}
}

extension Node: CustomStringConvertible {
	
	public var description: String {
		return "Node with value of \(value)"
	}
}

public extension Node where Node.Element: Equatable {
	
	func nodeValueEqualTo(valueOf aNode: Node) -> Bool {
		return self.value == aNode.value
	}
}
```